src/adapters/poi/idunn_poi.js

```diff
@@ -86,28 +86,35 @@ export default class IdunnPoi extends Poi {
     }
   }
 
-  static async poiApiLoad(id, options = {}) {
+  static async poiApiLoad(obj, options = {}) {
     let rawPoi = null;
-    const url = `${serviceConfig.idunn.url}/v1/places/${id}`;
+    const url = `${serviceConfig.idunn.url}/v1/places/${obj.id}`;
     let requestParams = {};
     if (options.simple) {
       requestParams = {verbosity: 'short'};
     }
     try {
-      rawPoi = await Ajax.getLang(url, requestParams);
+      const headers = {};
+      if (obj.suggestionRank !== undefined && obj.query !== undefined) {
+        if (obj.position !== undefined) {
```

Comment:
actually, we will need more context, such as the language used in the query.  

It might be cleaner to define a `queryContext` object that could be passed in the `options` object, instead of this cumbersome validation ?

Reply:
Indeed.